modules/templates/helper.go

```diff
@@ -622,6 +624,28 @@ func Str2html(raw string) template.HTML {
 	return template.HTML(markup.Sanitize(raw))
 }
 
+// RenderMarkup render markdown/markup specified in type to sanitized HTML
+func RenderMarkup(markupType, raw string) template.HTML {
+	var err error
+	var renderedContent string
+
+	if markupType == "markdown" {
+		if renderedContent, err = markdown.RenderString(&markup.RenderContext{}, raw); err != nil {
+			log.Warn("RenderMarkup: Invalid markdown? %v", err)
+			return template.HTML(html.EscapeString(raw))
+		}
+		return template.HTML(markup.Sanitize(renderedContent))
+	} else if markupType == "markup" {
+		if renderedContent, err = markup.RenderString(&markup.RenderContext{}, raw); err != nil {
```

Comment:
This has already included markdown rendering.

Reply:
And it seems we just needs `markdown` here.

Reply:
It was requested by sliverwind . https://github.com/go-gitea/gitea/pull/17980#pullrequestreview-834104385 

Personally I also prefer to keep it simple and only introduce one render.

Reply:
Yeah, I guess we can actually simplify and only support markdown, at least until we can detect other types (like RST) based on entered content.

By the way, is there speaking anything against just using [Str2html](https://github.com/go-gitea/gitea/blob/f48771ae78b5e5b3caa60c55b3253229ea804ab5/modules/templates/helper.go#L620-L623)?

Reply:
AFAIR it didn't render markdown tables

Reply:
Yeah seems like `Str2html` is mislabeled and actually deals renders HTML strings into template.HMTL, so `markdown.RenderString` seems fine.